src/session_impl.cpp

```diff
@@ -1887,6 +1864,50 @@ namespace {
 		}
 	}
 
+	void session_impl::remove_listen_sockets(std::vector<listen_endpoint_t>& eps)
+	{
+		m_listen_sockets.remove_if([&eps, this](listen_socket_t const& sock)
```

Comment:
This is the special `std::list` version of `remove_if` which removes the items as it goes.